libs/langgraph/src/pregel/types.ts

```diff
@@ -40,38 +71,83 @@ export type StreamMode =
+      SingleStreamModeOutput<Mode[2], Nodes, Channels>
     ]
   : never;
 
+// Gross hack to avoid recursion limits & define the narrowest type that covers all possible stream output types
+// This is what gets passed to the lower abstraction layers (such as `Runnable`) and the precise output type is
+// narrowed at higher levels such as `StateGraph.stream()` to a specific one once the `StreamMode` is known
+export type AllStreamOutputTypes<
+  Nodes extends NodesType,
+  Channels extends ChannelsType
+> =
+  | StreamOutput<["values"], Nodes, Channels>
+  | StreamOutput<["updates"], Nodes, Channels>
+  | StreamOutput<["debug"], Nodes, Channels>
+  | StreamOutput<["values", "updates"], Nodes, Channels>
+  | StreamOutput<["values", "values"], Nodes, Channels>
+  | StreamOutput<["values", "debug"], Nodes, Channels>
+  | StreamOutput<["updates", "updates"], Nodes, Channels>
+  | StreamOutput<["updates", "values"], Nodes, Channels>
+  | StreamOutput<["updates", "debug"], Nodes, Channels>
+  | StreamOutput<["debug", "updates"], Nodes, Channels>
+  | StreamOutput<["debug", "values"], Nodes, Channels>
+  | StreamOutput<["debug", "debug"], Nodes, Channels>
+  | StreamOutput<["updates", "updates", "updates"], Nodes, Channels>
+  | StreamOutput<["updates", "updates", "values"], Nodes, Channels>
+  | StreamOutput<["updates", "updates", "debug"], Nodes, Channels>
+  | StreamOutput<["updates", "values", "updates"], Nodes, Channels>
+  | StreamOutput<["updates", "values", "values"], Nodes, Channels>
+  | StreamOutput<["updates", "values", "debug"], Nodes, Channels>
+  | StreamOutput<["updates", "debug", "updates"], Nodes, Channels>
+  | StreamOutput<["updates", "debug", "values"], Nodes, Channels>
+  | StreamOutput<["updates", "debug", "debug"], Nodes, Channels>
+  | StreamOutput<["values", "updates", "updates"], Nodes, Channels>
+  | StreamOutput<["values", "updates", "values"], Nodes, Channels>
+  | StreamOutput<["values", "updates", "debug"], Nodes, Channels>
+  | StreamOutput<["values", "values", "updates"], Nodes, Channels>
+  | StreamOutput<["values", "values", "values"], Nodes, Channels>
+  | StreamOutput<["values", "values", "debug"], Nodes, Channels>
+  | StreamOutput<["values", "debug", "updates"], Nodes, Channels>
+  | StreamOutput<["values", "debug", "values"], Nodes, Channels>
+  | StreamOutput<["values", "debug", "debug"], Nodes, Channels>
+  | StreamOutput<["debug", "updates", "updates"], Nodes, Channels>
+  | StreamOutput<["debug", "updates", "values"], Nodes, Channels>
+  | StreamOutput<["debug", "updates", "debug"], Nodes, Channels>
+  | StreamOutput<["debug", "values", "updates"], Nodes, Channels>
+  | StreamOutput<["debug", "values", "values"], Nodes, Channels>
+  | StreamOutput<["debug", "values", "debug"], Nodes, Channels>
+  | StreamOutput<["debug", "debug", "updates"], Nodes, Channels>
+  | StreamOutput<["debug", "debug", "values"], Nodes, Channels>
+  | StreamOutput<["debug", "debug", "debug"], Nodes, Channels>;
```

Comment:
@nfcampos I wrote this monstrosity to union all the different possible permutations of of `StreamMode` being provided. I'm still running into issues where the output of `stream` (a `StreamOutput` instantiation) isn't assignable to `AllStreamOutputTypes`). Any ideas?